code/modules/mob/living/carbon/xenomorph/castes/runner/mutations_runner.dm

```diff
@@ -0,0 +1,461 @@
+		evasion.alternate_action_activate()
+	return ..()
+
+/datum/mutation_upgrade/shell/upfront_evasion/on_structure_update(previous_amount, new_amount)
+	. = ..()
+	if(!.)
+		return
+	var/datum/action/ability/xeno_action/evasion/evasion = xenomorph_owner.actions_by_path[/datum/action/ability/xeno_action/evasion]
+	if(!evasion)
+		return
+	evasion.evasion_starting_duration += get_duration(new_amount - previous_amount)
+
+/// Returns the amount of seconds that is added to Evasion's starting duration.
+/datum/mutation_upgrade/shell/upfront_evasion/proc/get_duration(structure_count)
+	return duration_per_structure * structure_count
+
+/datum/mutation_upgrade/shell/borrowed_time
+	name = "Borrowed Time"
+	desc = "Your critical threshold is decreased by 100. While you have negative health, you are staggered and cannot slash attack. If you have negative health for more than 2/3/4s, your critical threshold is increased back until you reach full health."
+	/// For the first structure, the amount of deciseconds that they can keep the critical threshold once they get negative health.
+	var/duration_initial = 1 SECONDS
+	/// For each structure, the amount of deciseconds that they can keep the critical threshold once they get negative health.
+	var/duration_per_structure = 1 SECONDS
+	/// Has the critical threshold been increased already?
+	var/critical_threshold_boosted = FALSE
+	/// The amount to decrease the critical threshold for.
+	var/critical_threshold_amount = 100
+	/// The timer that will reverse the critical threshold.
+	var/critical_threshold_timer
+
+/datum/mutation_upgrade/shell/borrowed_time/get_desc_for_alert(new_amount)
+	if(!new_amount)
+		return ..()
+	return "Your critical threshold is decreased by [critical_threshold_amount]. While you have negative health, you are staggered and cannot slash attack. If you have negative health for more than [get_duration(new_amount) * 0.1] seconds, your critical threshold is increased back until you reach full health."
+
+/datum/mutation_upgrade/shell/borrowed_time/on_mutation_enabled()
+	RegisterSignals(xenomorph_owner, list(COMSIG_XENOMORPH_BRUTE_DAMAGE, COMSIG_XENOMORPH_BURN_DAMAGE), PROC_REF(on_damage))
+	if(!critical_threshold_boosted && xenomorph_owner.health >= xenomorph_owner.maxHealth)
+		toggle(TRUE)
+	return ..()
+
+/datum/mutation_upgrade/shell/borrowed_time/on_mutation_disabled()
+	UnregisterSignal(xenomorph_owner, list(COMSIG_XENOMORPH_BRUTE_DAMAGE, COMSIG_XENOMORPH_BURN_DAMAGE))
+	if(critical_threshold_timer)
+		reverse_critical_threshold()
+	else if(critical_threshold_boosted)
+		toggle()
+	return ..()
+
+/// Increases or decreases the critical threshold amount for the owner.
+/datum/mutation_upgrade/shell/borrowed_time/proc/toggle(silent = FALSE)
+	critical_threshold_boosted = !critical_threshold_boosted
+	if(critical_threshold_boosted)
+		if(!silent)
+			to_chat(xenomorph_owner, span_notice("You feel like you're ready to be on Borrowed Time."))
```

Comment:
balloon alert this, tochats are for things we need to know later and or low priority